### ISSUE TITLE
Can't add a second header row

### ISSUE THREAD

**Issue by Artur-** [vaadin/vaadin-grid-flow#59]
Use case, something like
```
|-------------------|
| Year 2018         |
|-------------------|
| Q1 | Q2 | Q3 | Q4 |
|-------------------|
```

Either the API for adding the "Year 2018" header is missing or it is too well hidden

**Comment by pleku** [comment]
The docs are there but could benefit from some clarification and mayhaps even an image of the end result:
https://github.com/vaadin/flow/blob/master/flow-documentation/components/tutorial-flow-grid.asciidoc#grouping-columns

There is a demo but it has not been deployed to the vcom site yet. @Artur- were you missing it completely or just unable to find it or thought it was doing something else? 
https://github.com/vaadin/vaadin-grid-flow/blob/master/src/test/java/com/vaadin/flow/component/grid/demo/GridView.java#L635

**Comment by Artur-** [comment]
You are kidding about the API, right? ;)

Maybe I'm special but I would in no way start by merging two colums to create a grouped header in a separate header row. If I merge two columns in a grid, I would expect the columns to be merged and the original columns disappear.

**Comment by pleku** [comment]
>Maybe I'm special but I would in no way start by merging two colums to create a grouped header in a separate header row. If I merge two columns in a grid, I would expect the columns to be merged and the original columns disappear.

IIRC the team looked at the _old way_ from FW 8 `Grid` and they didn't like/understand that one, and ended up using this one. The whole concept of having `rows` in the header/footer is gone now and it is very different from the old in that sense.

I think we can make it better, but at this point would like to get more feedback. You're probably the first one to 1) discover the API 2) try it.

**Comment by gilberto-torrezan** [comment]
I agree that the name `mergeColumns` is a bit misleading. Maybe a better name for it would solve the issue (`groupColumns` ?).

The whole API design was made to make it closer to the API use by the element itself. They have a special webcomponent to group columns (called `vaadin-column-group`), and it's just a matter of adding or removing columns to the group to make them look grouped in the final Grid.

Examples: https://cdn.vaadin.com/vaadin-grid/5.0.0-alpha6/demo/#grid-columns-demos

**Comment by mvysny** [comment]
With Vaadin 8, I could add another header row and populate it with filtering components. The recommended Vaadin 10 way is to group columns, but I do not want to group the columns - I want to add a filtering component below the existing header.

**Comment by pleku** [comment]
>I could add another header row and populate it with filtering components.

First we should build a demo for this as an example, and then apply #ffs for making that as easy and intuitive as possible.

**Comment by mvysny** [comment]
Found a way to add component *above* the header:
```kotlin
fun Grid.Column<*>.appendHeaderAbove(header: Component) {
    val index = grid.element.indexOfChild(element)
    grid.element.removeChild(element)
    val columnGroup = ColumnGroup(grid, this)
    grid.element.insertChild(index, columnGroup.element)
    columnGroup.setHeader(header)
}
```
Very hacky and fragile. I would definitely appreciate a ffs way provided by Vaadin directly.

**Comment by gilberto-torrezan** [comment]
@mvysny One way of fixing this is to provide a `ComponentRenderer` or a `TemplateRenderer` for the header that contains both the header text and the filtering text field.

But I agree that the public API should contain a proper way of doing this.

Created https://github.com/vaadin/vaadin-grid-flow/issues/120

**Comment by pleku** [comment]
Oops, wrong issue

**Comment by pekam** [comment]
Fixed in #163
